File: videoLocadoraSpring/src/main/java/com/videoLocadora/domain/controle_acervo/Titulo.java
package com.videoLocadora.domain.controle_acervo;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import lombok.Data;

@Data
@Entity
public class Titulo {

    @Id
    private Long id;

    @Column(length = 255, nullable = false)
    private String nome;

    @Column(nullable = false)
    private String ano;

    @Column(nullable = false)
    private String sinopse;

    @Column(nullable = false)
    private String categoria;
    
}

File: videoLocadora/src/app/type/titulo.ts
import { Ator } from "./ator";
import { Classe } from "./classe";
import { Diretor } from "./diretor";

export interface Titulo {
    _id: number;
    nome: string;
    ano: number;
    sinopse: string;
    categoria: string;
    ator: Array<Ator>;
    diretor: Diretor;
    classe: Classe;
}